app/controllers/analysis_page_finder_controller.rb

```diff
@@ -0,0 +1,22 @@
+class AnalysisPageFinderController < ApplicationController
+  skip_before_action :authenticate_user!
+  include AnalysisPages
+
+  def show
+    urn = params[:urn]
+    analysis_class = params[:analysis_class]
+
+    school = School.find_by(urn: urn)
+    alert_type = AlertType.find_by(class_name: analysis_class)
+
+    if school && school.latest_analysis_pages.any?
+      analysis_page = school.latest_analysis_pages.includes(:alert).detect { |page| page.alert.alert_type_id == alert_type.id }
+    end
+
+    if analysis_page
+      redirect_to school_analysis_path(school_id: school.slug, id: analysis_page.id)
```

Comment:
`school_analysis_path(schoo, analysis_page)`?

Reply:
Yes, of course!